frontend/src/scenes/userLogic.tsx

```diff
@@ -98,5 +118,38 @@ export const userLogic = kea<userLogicType<UserType, EventProperty>>({
                 actions.userUpdateFailure(error, updateKey)
             }
         },
+        createPersonalAPIKeyRequest: async ({ label }) => {
+            let newKey
+            try {
+                newKey = await api.create('api/personal_api_key/', { label })
+            } catch (e) {
+                actions.createPersonalAPIKeyFailure(label)
+                return
+            }
+            const user = selectors.user()
+            actions.createPersonalAPIKeySuccess(user, newKey)
+        },
+        createPersonalAPIKeySuccess: ({ key }) => {
+            toast(<div className="text-success">Personal API key "{key.label}" successfully created</div>)
+        },
+        createPersonalAPIKeyFailure: ({ label }: { label: string }) => {
+            toast(<div className="text-danger">Could not create personal API key "{label}"</div>)
+        },
+        deletePersonalAPIKeyRequest: async ({ key }: { key: PersonalAPIKeyType }) => {
+            try {
+                await api.delete(`api/personal_api_key/${key.id}/`)
+            } catch (e) {
+                actions.deletePersonalAPIKeyFailure(key)
+                return
+            }
+            const user = selectors.user()
+            actions.deletePersonalAPIKeySuccess(user, key)
+        },
+        deletePersonalAPIKeySuccess: ({ key }) => {
```

Comment:
Because `key.label` is the only property used, consider:
```
deletePersonalAPIKeySuccess(label)
```

Unless more properties of `key` are planning on being used in the future _or_ has better abstractional integrity.

Reply:
I recommend a read through https://kea.js.org/docs/guide/concepts/. It covers always passing payloads in the form of objects.

Reply:
The suggestion wasn't necessarily about wrapping the object (the `key`) in another object. It was about restricting the amount of information available to the function:
```
deletePersonalAPIKeySuccess({ label })
```
would also be fine, but I'll check out the docs.